1.data-exploration/scripts/nbconverted/1.model_data_visualization.py

```diff
@@ -150,69 +148,116 @@
         age_distribution.append(np.nan)
 
 
+# In[9]:
+
+
+# Add columns age_categories & age_distribution
+model_df = model_df.assign(
+    age_categories=age_categories,
+    age_distribution=age_distribution
+)
+
+# Output file
+model_df.to_csv(model_ouput_age_cleaned_file, index=False)
+
+print(model_df.shape)
+model_df.head(3)
+
+
+# ## Visualize age categories and distribution
+
 # In[10]:
 
 
-# New dataframe containing two new columns age_categories & age_distribution
-df_age_visual = df_model.assign(
-    age_categories=age_categories, age_distribution=age_distribution
+age_categories_bar = (
+    gg.ggplot(model_df, gg.aes(x="age_categories")) + gg.geom_bar()
 )
 
-df_age_visual.head()
+age_categories_bar.save(age_category_output_figure, dpi=500)
+
+age_categories_bar
 
 
 # In[11]:
 
 
-# save the new data frame to a new .csv file in 0.data -download module
-df_save_destination = pathlib.Path(
-    "../0.data-download/data/Model_age_column_cleaned.csv"
+age_distribution_plot = (
+    gg.ggplot(model_df, gg.aes(x="age_distribution"))
+    + gg.geom_density()
+    + gg.geom_vline(xintercept=adult_threshold, linetype="dashed", color="red")
 )
-df_age_visual.to_csv(df_save_destination, index=False)
+
+age_distribution_plot.save(age_distribution_output_figure, dpi=500)
+
+age_distribution_plot
 
 
 # In[12]:
 
 
-print(df_age_visual.shape)
-df_age_visual.head()
+pd.DataFrame(age_categories).loc[:, 0].value_counts()
 
 
 # In[13]:
 
 
-age_categories_bar = (
-    gg.ggplot(df_age_visual, gg.aes(x="age_categories")) + gg.geom_bar()
+gendersamp_plot = (
+    gg.ggplot(model_df, gg.aes(x="sex")) + gg.geom_bar()
 )
-print(age_categories_bar)
-acb_output = pathlib.Path("./figures/age_categories_bar_chart.png")
-age_categories_bar.save(acb_output)
+    
+gendersamp_plot.save(sex_output_figure)
+
+gendersamp_plot
 
 
+# ## What cell lines are pediatric cancer?
+
 # In[14]:
 
 
-age_distribution_plot = (
-    gg.ggplot(df_age_visual, gg.aes(x="age_distribution"))
-    + gg.geom_density()
-    + gg.geom_vline(xintercept=adult_threshold, linetype="dashed", color="red")
+pediatric_model_df = (
+    model_df
+    .query("age_categories == 'Pediatric'")
+    .query("DepMap_ID in @sample_overlap")
+    .reset_index(drop=True)
```

Comment:
I don't see anything that makes me concerned about the formatting more generally, but the multi-line programming in this notebook made me wonder about formatting tools which might be helpful if they're not already in use. Sharing along [`black[jupyter]`](https://black.readthedocs.io/en/stable/getting_started.html#installation) and [`jupyterlab_code_formatter`](https://github.com/ryantam626/jupyterlab_code_formatter), which can be helpful depending on your Jupyter IDE and development environment.

Reply:
yep, good point. I haven't ever done this before, but it's about time that I start!

I ended up going with jupyterlab_code_formatter, and made corresponding changes in the environment file.